moped-editor/src/views/projects/projectView/ProjectComponents.js

```diff
@@ -183,9 +151,14 @@ const ProjectComponents = () => {
     <ApolloErrorHandler errors={error}>
       {componentEditMode && (
         <ProjectComponentEdit
-          componentId={selectedComp}
+          selectedProjectComponent={selectedProjectComponent}
+          selectedComponentFeatureCollection={selectedComponentFeatureCollection}
           handleCancelEdit={handleCancelEdit}
-          projectFeatureCollection={featureFullCollection}
+          projectFeatureCollection={projectFeatureCollection}
+          mopedComponents={data?.moped_components || []}
+          mopedSubcomponents={data?.moped_subcomponents || []}
+          lineRepresentable={lineRepresentable}
+          availableTypes={availableTypes}
```

Comment:
now passing many more props to `ProjectComponentEdit`, which will save many renders. probably a good use case for `Context` but....i don't like `Context`?

i'm not familiar enough with Apollo—but perhaps some of this data could be fetched from cache from within the `ProjectComponentEdit`?

not going to overthink it—this seems fine